### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Mikhail-Pashkevich**

### CODE REVIEW

File: battle-service/src/main/java/by/pashkevich/mikhail/model/entity/Battle.java [hunk 1]
```diff
@@ -37,4 +39,35 @@ public class Battle {
...
+
+    public void setPlayerOnEmptyPlace(User player) {
+        if (playerO == null) {
+            playerO = player;
+        } else {
+            playerX = player;
+        }
+    }
+
+    public Value getValueByUserId(Long userId) {
```
Comment (Volad) [inline battle-service/src/main/java/by/pashkevich/mikhail/model/entity/Battle.java:60]: use strings as user id's, in uuid format

File: battle-service/src/main/java/by/pashkevich/mikhail/service/impl/BattleServiceImpl.java [hunk 2]
```diff
@@ -29,72 +31,69 @@ public class BattleServiceImpl implements BattleService {
...
-
-        return battle;
+        return battleRepository.save(battle);
     }
 
     @Override
-    public Battle join(Long playerId) {
-        User player = userService.getById(playerId);
+    public Battle join() {
+        User player = userService.getAuthenticatedUser();
```
Comment (Volad) [inline battle-service/src/main/java/by/pashkevich/mikhail/service/impl/BattleServiceImpl.java:49]: authentication hould not be involved into services layers

File: battle-service/src/main/java/by/pashkevich/mikhail/model/entity/Battle.java [hunk 3]
```diff
@@ -37,4 +39,35 @@ public class Battle {
...
+        if (playerX.getId().equals(userId)) {
+            return Value.VALUE_X;
+        } else if (playerO.getId().equals(userId)) {
+            return Value.VALUE_O;
+        } else {
+            throw new IncorrectDataException("Can't process userId: " + userId);
+        }
+    }
+
+    public boolean isExist(User player) {
```
Comment (Volad) [inline battle-service/src/main/java/by/pashkevich/mikhail/model/entity/Battle.java:70]: i would avoid such methods.
you can peform such checks in a service

File: battle-service/src/main/java/by/pashkevich/mikhail/service/impl/BattleServiceImpl.java [hunk 4]
```diff
@@ -29,72 +31,69 @@ public class BattleServiceImpl implements BattleService {
...
     @Override
-    public Battle makeMove(Long battleId, Integer step, Value value) {
-        Battle battle = battleRepository.findById(battleId).orElseThrow(() -> {
-            throw new UnsupportedOperationException("Not implemented yet!");
-        });
+    public Battle makeMove(Long battleId, Step step) {
+        Battle battle = battleRepository.getReferenceById(battleId);
+
+        if (!battle.getBattleStatus().isActiveBattleStatus()) {
+            throw new IncorrectDataException("Battle with id = " + battleId + " unavailable");
```
Comment (Volad) [inline battle-service/src/main/java/by/pashkevich/mikhail/service/impl/BattleServiceImpl.java:74]: use strng format, to format error message, or better create own exception ie BattleUnavailableException

File: battle-service/src/main/java/by/pashkevich/mikhail/service/impl/BattleServiceImpl.java [hunk 5]
```diff
@@ -29,72 +31,69 @@ public class BattleServiceImpl implements BattleService {
...
+        }
 
-        if (FINISHED.equals(battle.getBattleStatus()) || WAIT_FOR_PLAYER.equals(battle.getBattleStatus())) {
-            return battle;
+        User user = userService.getAuthenticatedUser();
```
Comment (Volad) [inline battle-service/src/main/java/by/pashkevich/mikhail/service/impl/BattleServiceImpl.java:77]: move authentication and security check to separate service and do these check prior executing service logic.

File: user-service/src/main/java/by/pashkevich/mikhail/model/User.java [hunk 6]
```diff
@@ -33,4 +34,12 @@ public class User {
 
     @Column
     private String nickname;
+
+
+    public void addRole(Role role) {
```
Comment (Volad) [inline user-service/src/main/java/by/pashkevich/mikhail/model/User.java:39]: i believe you don't need this method and part of public api. better to do this in appropriate service

File: battle-service/src/main/java/by/pashkevich/mikhail/model/entity/Battle.java (hunk 1 above)
Comment (Mikhail-Pashkevich) [reply battle-service/src/main/java/by/pashkevich/mikhail/model/entity/Battle.java:60]: I read about uuid, but I'm not sure that it makes sense to change something

File: battle-service/src/main/java/by/pashkevich/mikhail/model/entity/Battle.java (hunk 3 above)
Comment (Mikhail-Pashkevich) [reply battle-service/src/main/java/by/pashkevich/mikhail/model/entity/Battle.java:70]: I move this methods to BattleUtil class. See https://github.com/Mikhail-Pashkevich/PetProject/pull/6

File: battle-service/src/main/java/by/pashkevich/mikhail/service/impl/BattleServiceImpl.java (hunk 2 above)
Comment (Mikhail-Pashkevich) [reply battle-service/src/main/java/by/pashkevich/mikhail/service/impl/BattleServiceImpl.java:49]: I decide to provide User as args to method and I created AuthAspect with custom annotation to realized it. See https://github.com/Mikhail-Pashkevich/PetProject/pull/6

File: battle-service/src/main/java/by/pashkevich/mikhail/service/impl/BattleServiceImpl.java (hunk 4 above)
Comment (Mikhail-Pashkevich) [reply battle-service/src/main/java/by/pashkevich/mikhail/service/impl/BattleServiceImpl.java:74]: done in https://github.com/Mikhail-Pashkevich/PetProject/pull/6

File: battle-service/src/main/java/by/pashkevich/mikhail/service/impl/BattleServiceImpl.java (hunk 5 above)
Comment (Mikhail-Pashkevich) [reply battle-service/src/main/java/by/pashkevich/mikhail/service/impl/BattleServiceImpl.java:77]: See BattleVerifyService in https://github.com/Mikhail-Pashkevich/PetProject/pull/6

File: battle-service/src/main/java/by/pashkevich/mikhail/model/entity/Battle.java (hunk 1 above)
Comment (Volad) [reply battle-service/src/main/java/by/pashkevich/mikhail/model/entity/Battle.java:60]:
```
@Id
    @Column(columnDefinition = "uuid")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
            name = "UUID",
            strategy = "org.hibernate.id.UUIDGenerator"
    )
    private String id;
```
you decide.

**Review by Volad — APPROVED** [review]

**Pull request merged**